hashbang/hashbang
# initial 'make test' support

Setup for integration testing across all #! core infra via 'make test'

Currently just spins everything up, and executes 'ls' over ssh to the shellbox as root and checks ssh is up and happy on signup-ssh.

All test dependencies live in a separate test container where it is okay to be dirty. Only dependency for testing should be semi recent "docker-compose' thanks to a semi-batshit docker-in-docker pattern.

This boilerplate should enable fairly straight forward testing of things like:
- test loading up postgres userdb schemas
- create/test read-replica on shellbox
- directly create dummy users in db
- test ssh to dummy users
- verify dotfiles are setup properly
- sanity check raw CRUD queries against db
- clone/test admin-tools
- postfix container setup & testing
- test adding users end-to-end as new libraries are created.
- simulate network latency and failure modes across entire cluster via Entropy (https://github.com/buildertools/entropy)
- test netsplit handling with irc servers
